fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -994,9 +1002,8 @@ where
         self.memory.write(self.owner, dst_addr, length)?;
         self.input_contracts.check(&contract_id)?;
 
-        let contract = super::contract::contract(self.storage, &contract_id)?;
-        let contract_bytes = contract.as_ref().as_ref();
-        let contract_len = contract_bytes.len();
+        let contract_bytes = self.memory.write(self.owner, dst_addr, length)?;
+        let contract_len = contract_size(&self.storage, &contract_id)?;
         let charge_len = core::cmp::max(contract_len as u64, length);
```

Comment:
We never read more than `length` bytes with the new version. Should `charge_len` be changed accordingly to reflect this?

Reply:
I also prefer to keep the current charging strategy, even if we overcharge user=) The `length` in most cases should be not more than `contract_len`.